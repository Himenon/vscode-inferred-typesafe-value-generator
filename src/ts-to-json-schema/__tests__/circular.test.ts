import * as path from "path";
import * as Module from "../index";

const basePath = path.join(__dirname, "./sample-code");

describe("Popular JSON Schema Test", () => {
  test("Load files directly", () => {
    const schema = Module.generateJsonSchema({
      files: [path.join(basePath, "./circular-dependency.ts")],
      typeName: {
        isUnknown: false,
        firstTypeName: "Directory",
        isArray: false,
        intersectinoTypeNames: [],
        isIntersectionType: false,
      },
      basePath,
    });
    expect(schema).toMatchSnapshot();
  });
});
