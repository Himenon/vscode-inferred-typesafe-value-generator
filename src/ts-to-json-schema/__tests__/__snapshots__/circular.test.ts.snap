// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular JSON Schema Test Load files directly 1`] = `
Object {
  "jsonSchema": Object {
    "$ref": "#/definitions/Directory",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": Object {
      "Child": Object {
        "anyOf": Array [
          Object {
            "$ref": "#/definitions/File",
          },
          Object {
            "$ref": "#/definitions/Directory",
          },
        ],
      },
      "Directory": Object {
        "additionalProperties": false,
        "properties": Object {
          "children": Object {
            "items": Object {
              "$ref": "#/definitions/Child",
            },
            "type": "array",
          },
        },
        "required": Array [
          "children",
        ],
        "type": "object",
      },
      "File": Object {
        "additionalProperties": false,
        "properties": Object {
          "name": Object {
            "type": "string",
          },
        },
        "required": Array [
          "name",
        ],
        "type": "object",
      },
    },
  },
}
`;
